crates/storage/libmdbx-rs/src/cursor.rs

```diff
@@ -59,6 +60,14 @@ where
         self.cursor
     }
 
+    /// Returns an Iterator over key value slices
+    ///
+    /// Note: The lifetime ensures that the transaction is kept alive while entries are used
+    #[allow(clippy::should_implement_trait)]
+    pub fn into_iter<'cur>(self) -> IntoIter<'cur, K, Cow<'cur, [u8]>, Cow<'cur, [u8]>> {
```

Comment:
... and then you can implement the trait

This `'cur` can be `'static` which is not valid

Reply:
can't implement the trait to return values that are only valid for the existence of the transaction

but this isn't even used so I'm considering removing it entirely

Reply:
Ah the returned data is generic over `TableObject` which is deserialized from Tx, so the lifetime is that of Tx

Reply:
yeah I removed the lifetime from tableobject but this means we can no longer restrict it via that,

we only call `into_iter` once here:

https://github.com/paradigmxyz/reth/blob/91034b6cdaa89cf747139e1f0ca8edaff3dbcea5/crates/storage/libmdbx-rs/src/environment.rs#L211-L211

there's no easy way to provide these guarantees with the IntoIterator trait.
so this change seems fine to me, given that we only call this once

Reply:
In this case you can make the impl generic over `Key, Value`

Reply:
right, but I guess the motivation here was to iterate over slices directly, but can simply add another function for this

Reply:
My main goal here is to remove the clippy allow :D
You can remove the lifetime from the phantomdata and impl IntoIterator

Reply:
hmm, unclear how this would work, skeptical generics for an IntoIter impl are possible here

feel free to do that separately